### PR TITLE
Add entry storage

### LINKED ISSUE #273: Create serialisation classes for EntryInfo, ResumeEntry, EntryDescription and EntryBook

### PR DESCRIPTION
**Pull request by anubh-v**
Fixes #273 
With this PR,
The user defined XML is subject to the following conditions:

- Category attribute is compulsory for each entry, and must be valid as defined by `Category` class
- Entries may have 0 tag elements, but if they are provided, all tag names must be valid as defined by `Tag` class
- Entries may omit the entryInfo element, but if provided, all 3 attributes (title, subheader, duration) **must** be present and valid as defined by `EntryInfo` class
- Entries may omit the entryDesc element, but if provided, there must be at least one bullet element
- The XML cannot contain duplicate entries

Pending implementation:
- Validation of the bullet elements. Currently, empty string and whitespace will be accepted as bullets.

Sample XML:

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>

<entrybook>
<entry category="work">
    <entryInfo title="Facebook Internship" subheader="Coffee Boy" duration="Summer 2011"/>
    <entryDesc>
      <bullet>Implemented a scalable system of coffee delivery to mid and upper level executives. </bullet>
      <bullet> Personally oversaw daily coffee delivery to more than 20 remote and on-site teams. </bullet>
      <bullet> Went beyond my duties to provide additional donut delivery services. </bullet>
    </entryDesc>
    <tags>leadership</tags>
</entry>
<entry category="work">
    <entryInfo title="Carousell Internship" subheader="Donut Boy" duration="Summer 2010"/>
    <tags>donuts</tags>
    <tags>management</tags>
</entry>
<entry category="awards">
</entry>
</entrybook>
```

### CODE REVIEW

File: src/main/java/seedu/address/storage/entry/XmlAdaptedEntryDescription.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     * @throws IllegalValueException if there were any data constraints violated in the adapted entry description.
+     */
+    public EntryDescription toModelType() throws IllegalValueException {
+        EntryDescription desc = new EntryDescription();
+
+        if (bullets == null) {
+            throw new IllegalValueException(MESSAGE_MISSING_BULLETS);
+        }
+
+        for (String bullet : bullets) {
```
Comment (ongspxm) [inline src/main/java/seedu/address/storage/entry/XmlAdaptedEntryDescription.java]: wait... whr is the `bullets` intialized? is it just auto by the library?

File: src/main/java/seedu/address/storage/entry/XmlAdaptedEntryInfo.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            return true;
+        }
+
+        if (!(other instanceof XmlAdaptedEntryInfo)) {
+            return false;
+        }
+
+        XmlAdaptedEntryInfo otherAdaptedEntryInfo = (XmlAdaptedEntryInfo) other;
+        return Objects.equals(title, otherAdaptedEntryInfo.title)
+                 && Objects.equals(subheader, otherAdaptedEntryInfo.subheader)
```
Comment (ongspxm) [inline src/main/java/seedu/address/storage/entry/XmlAdaptedEntryInfo.java]: lolz, any reason why the change in style for this? instead of `subheader.equals(...)`? i mean, i know it works and all, but... consistency?

Comment (anubh-v) [reply src/main/java/seedu/address/storage/entry/XmlAdaptedEntryInfo.java]: Thanks for raising this.
The AB4 code actually uses this style in `XmlAdaptedPerson`, which is why I adopted it.

As for the reason, I believe the objective is to perform the `equals` check in a null - safe way. 
The `equals` check will work if either argument is null, whereas `subheader.equals(..)` would throw an exception.

These fields may be null if the user does not provide the corresponding XML element or attribute. For example, `subheader` will be null if no subheader attribute is provided.

File: src/main/java/seedu/address/storage/entry/XmlAdaptedEntryDescription.java (hunk 1 above)
Comment (anubh-v) [reply src/main/java/seedu/address/storage/entry/XmlAdaptedEntryDescription.java]: The expectation is that the XML has been parsed by the JAXB framework before `toModelType` is run. 
This would mean either `bullets` is initialised as a list with at least 1 element, or is null (because user did not provide any bullet elements in the XML)

Comment (anubh-v) [reply src/main/java/seedu/address/storage/entry/XmlAdaptedEntryDescription.java]: The expectation is that the XML has been parsed by the JAXB framework before `toModelType` is run. 
This would mean either `bullets` is initialised as a list with at least 1 element, or is null (because user did not provide any bullet elements in the XML)

**Comment by anubh-v** [comment]
More tests to be added #264
